### ISSUE TITLE
Feature: Persist in REDIS or PostgreSQL Payments Done

### ISSUE THREAD

**Issue by jvxis** [alephium/mining-pool#58]
Hi, It would be nice to persist in Redis or PostgreSQL all the payments done successfully by the mining pool.

Thanks,
Jaime

**Comment by killerwhile** [comment]
Not sure if redis is the right place to put this, since it's not the best pattern to put unbounded collections in Redis, but storing the transactions in PostgresQL is a good idea, thanks for the proposal!

**Comment by jvxis** [comment]
Yes, I'm using both Redis and PostgreSQL combined for this Dashboard. Including a table in PostgreSQL with all transactions (Payments) made would be nice. 
![image](https://user-images.githubusercontent.com/108929149/210184214-72c3dfd7-5994-4575-be0e-04eb0ccdd94d.png)

**Comment by Lbqds** [comment]
As @killerwhile said, it might be better to store the payments in PostgreSQL.

But if you want this now, you can change [this line](https://github.com/alephium/mining-pool/blob/5f1cc5c48501adc4e9f6c37c74f43568cd63c46b/lib/paymentProcessor.js#L436) to `redisTx.expire(_this.txRewardsKey(txId), TIMEOUT)`, instead of removing the succeed payments immediately, it saves payments on Redis until timeout.

**Comment by jvxis** [comment]
Thanks. Does this will persist on REDIS the hash, miner address and paid amount?

**Comment by Lbqds** [comment]
> Thanks. Does this will persist on REDIS the hash, miner address and paid amount?

Yes, the hset key is `rewards: txId`, the entry key is the miner address, and the entry value is paid amount.

**Comment by jvxis** [comment]
Thanks, I will try. @Lbqds Do you know the amount of time I can set this timeout? Can it be one day?

**Comment by killerwhile** [comment]
Yes it can be a day, or a week, in seconds, as you need.
